### PR TITLE
Multiple bug fixes

### PR DESCRIPTION
**Pull request by ye-luo**

### CODE REVIEW

File: src/QMCHamiltonians/QMCHamiltonian.cpp [hunk 1]
```diff
@@ -515,11 +515,9 @@ void QMCHamiltonian::flex_evaluate(const RefVector<QMCHamiltonian>& H_list,
         op.setObservables(ham.Observables);
         op.setParticlePropertyList(pset.PropertyList, ham.myIndex);
       };
-      HC_list[i].get().mw_evaluate(HC_list, P_list);
+      HC_list[0].get().mw_evaluate(HC_list, P_list);
```
Comment (PDoakORNL) [inline src/QMCHamiltonians/QMCHamiltonian.cpp:518]: This is all pretty confusing. That was my thinking documenting via the int what the loop is over, although I grabbed the wrong size. The extract_HC_list is doing what I was thinking of here with the HC_list[i] although with the right limit this should be valid as well. Accessing the mv_evaluate through a particular OperatorBase is just a hack to get at the VTable. 

Anyway this is fine no need to change.

File: src/QMCWaveFunctions/TrialWaveFunction.cpp [hunk 2]
```diff
@@ -162,19 +162,19 @@ void TrialWaveFunction::flex_evaluateLog(const RefVector<TrialWaveFunction>& wf_
...
+    // Ye: temporal workaround to have P.G/L always defined.
+    // remove when KineticEnergy use WF.G/L instead of P.G/L
+    for (int iw = 0; iw < wf_list.size(); iw++)
+      copyToP(p_list[iw], wf_list[iw]);
   }
   else if (wf_list.size() == 1)
   {
     wf_list[0].get().evaluateLog(p_list[0]);
   }
-  auto copyToP = [](ParticleSet& pset, TrialWaveFunction& twf){
```
Comment (PDoakORNL) [inline src/QMCWaveFunctions/TrialWaveFunction.cpp:170]: I moved this code block here to also catch the size 1 case.  the evaluateLog used to be followed by
```C++
WF_list[0]->G = P_list[0]->G;
WF_list[0]->L = P_list[0]->L;
```
But now I see this is not what that code was.

Comment (PDoakORNL) [reply src/QMCWaveFunctions/TrialWaveFunction.cpp:170]: i.e. this doesn't replace it, so what do we need after the evaluateLog

**Review by PDoakORNL — CHANGES_REQUESTED** [review]

**Review by PDoakORNL — APPROVED** [review]

File: src/QMCHamiltonians/QMCHamiltonian.cpp (hunk 1 above)
Comment (ye-luo) [reply src/QMCHamiltonians/QMCHamiltonian.cpp:518]: HC_list is one component, many walkers not one walker many components. index i is the loop over components. I got 2 walker 5 components and the old code crashes.
Regarding how to access mw_evaluate. Probabaly in Oct. Let us discuss it all the possible solution and find the best one.

File: src/QMCWaveFunctions/TrialWaveFunction.cpp (hunk 2 above)
Comment (ye-luo) [reply src/QMCWaveFunctions/TrialWaveFunction.cpp:170]: Once we remove P.G/L, all the related code needs to be inspected but probably just check every placee which doesn't compile.

**Pull request merged**